### PR TITLE
Refactor: new default branch main

### LINKED ISSUE #290: Fully migrate to new deployment model
- [x] Create a `main` branch, add branch protection rules similar to that of `dev`
 - [x] Rename references to `dev` branch to use `main`
 - [x] Add logic to tag `main` branch's Docker image as `dev`
 - [x] Update pipeline logic for getting deployment environment to consider `main` branch
 - [x] Remove logic for deriving Docker tag name from Git branches for test/prod
 - [x] Delete `test` and `prod` branch triggers from `azure-pipeline.yml` and `deploy.yml`
 - [x] Make `main` the default branch in the repo

## After close / merge of #414

 - [ ] Delete `dev`, `test`, and `prod` branches and branch protection rules
 - [ ] Update `main` branch protection to require the `test` check

### PR DESCRIPTION
**Pull request by thekaveman**
Closes #290 

See #290 for post-merge steps.

### CODE REVIEW

File: terraform/pipeline/workspace.py [hunk 1]
```diff
@@ -9,14 +9,14 @@
...
 # the branches that correspond to environments
-ENV_BRANCHES = ["dev", "test", "prod"]
+ENV_BRANCHES = ["main"]
 
 if REASON == "PullRequest" and TARGET in ENV_BRANCHES:
-    # it's a pull request against one of the environment branches, so use the target branch
-    environment = TARGET
+    # it's a pull request against main, this is for the dev environment
+    environment = "dev"
 elif REASON in ["IndividualCI", "Manual"] and SOURCE in ENV_BRANCHES:
```
Comment (thekaveman) [inline terraform/pipeline/workspace.py:17]: I think this is for a manual run? Check my logic here please

Comment (angela-tran) [reply terraform/pipeline/workspace.py:17]: 👍 Yeah, `Manual` is for a manual run. 

`IndividualCI` is for when a push is made to the refs specified in the pipeline's `trigger`s.

Comment (angela-tran) [reply terraform/pipeline/workspace.py:17]: To go with my suggestion of removing the `ENV_BRANCHES` variable, I think this line can be changed to:
```
elif REASON in ["IndividualCI", "Manual"] and SOURCE == "main":
```

**Review by angela-tran — COMMENTED** [review]
I have a minor suggestion for the workspace Python script. The rest of this looks good to me. Thank you so much @thekaveman !

**Comment by thekaveman** [comment]
@machikoyasuda please review today if you are able.

**Pull request merged**